tests/Type/WebMozartAssert/data/impossible-check-eq-not-eq.php

```diff
@@ -0,0 +1,40 @@
+<?php declare(strict_types=1);
+
+namespace WebmozartAssertImpossibleCheckEqNotEq;
+
+use DateTimeInterface;
+use stdClass;
+use Webmozart\Assert\Assert;
+
+function sameInstancesAreAlwaysEqual(stdClass $a, stdClass $b): void
+{
+	Assert::eq($a, $a); // will always evaluate to true
+	Assert::notEq($b, $b); // will always evaluate to false
+}
+
+function instancesOfTheSameTypeAreNotIdenticalButCouldBeEqual(stdClass $a, stdClass $b, stdClass $c, stdClass $d): void
+{
+	Assert::eq($a, $b);
+	Assert::notEq($c, $d);
+
+	Assert::eq(createStdClass(), createStdClass());
+	Assert::notEq(createStdClass(), createStdClass());
+}
+
+function instancesOfDifferentTypesAreNeverEqual(stdClass $a, DateTimeInterface $b, stdClass $c, DateTimeInterface $d, stdClass $e, stdClass $f, stdClass $g, stdClass $h): void
+{
+	// These don't report anything as PHPStan doesn't currently support loose comparison with ==
+	Assert::eq($a, $b);
+	Assert::notEq($c, $d);
+
+	Assert::eq($e, new stdClass()); // will always evaluate to true, should not report anything
```

Comment:
Please adjust the asserts to what actually makes sense, and not what PHPStan currently reports. I'm pretty sure this one "'Call to static method Webmozart\Assert\Assert::eq() with stdClass and stdClass will always evaluate to true.'" on line 30 doesn't make sense. There might be more like that.

Reply:
No idea why some errors `'Call to static method Webmozart\Assert\Assert::notEq() with 1 and true will always evaluate to false.'` actually work. It'd be great to find out which line in TypeSpecifier is responsible for that.

Reply:
Must be the parts that continue specification if left and right expr strings are different or there are variables involved? https://github.com/phpstan/phpstan-src/blob/master/src/Analyser/TypeSpecifier.php#L452 But I'm just guessing now tbh.